test/tools/router_check/router.h

```diff
@@ -66,7 +66,8 @@ class RouterCheckTool : Logger::Loggable<Logger::Id::testing> {
    * config file.
    * */
   static RouterCheckTool create(const std::string& router_config_file,
-                                const bool disable_deprecation_check);
+                                const bool disable_deprecation_check,
+                                const std::string& listener_name);
```

Comment:
It is unclear what the function should do if listener_name is and empty string (validate all listener or not to validate anything).

Reply:
With the current implementation there will be no route and the tests would fail. I agree a more explicit fail message would be more appropriated. Will change this.